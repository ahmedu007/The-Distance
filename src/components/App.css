a {
  font-family: Lato, "Helvetica Neue", Arial, Helvetica, sans-serif;
  font-size: 16px;
  color: black;
  font-weight: 525;
  line-height: 1.45em;
}

#img-zoom-in {
  transition: all 0.2s ease-in-out;
  -webkit-transition: all 0.2s ease-in-out;
  -moz-transition: all 0.2s ease-in-out;
  -ms-transition: all 0.2s ease-in-out;
  -o-transition: all 0.2s ease-in-out;
}

#img-zoom-in:hover {
  transform: scale(1.05);
  -webkit-transform: scale(1.05);
  -moz-transform: scale(1.05);
  -ms-transform: scale(1.05);
  -o-transform: scale(1.05);
}
