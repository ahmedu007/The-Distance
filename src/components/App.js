import React from "react";
import "./App.css";
import EventCard from "./EventCard";
import { Container, Pagination, Button } from "semantic-ui-react";
import Navbar from "./Navbar";
import Loading from "./Loading";

class App extends React.Component {
  constructor() {
    super();
    this.handleChange = this.handleChange.bind(this);
    this.handleBackClick = this.handleBackClick.bind(this);
    this.handleNextClick = this.handleNextClick.bind(this);
    this.state = {
      activePage: 1,
      loading: true
    };
  }

  componentDidMount() {
    this.setState({ activePage: this.props.activePage });
  }

  handleChange(event, { activePage }) {
    event.preventDefault();
    this.setState({
      activePage
    });
    this.props.getEvents(activePage);
  }

  handleBackClick(event) {
    event.preventDefault();
    let activePage = this.state.activePage - 1;
    this.props.getEvents(activePage);
    this.setState({
      activePage
    });
  }

  handleNextClick(event) {
    event.preventDefault();
    let activePage = this.state.activePage + 1;
    this.props.getEvents(activePage);
    this.setState({
      activePage
    });
  }

  render() {
    const { activePage } = this.props;
    return (
      <div>
        <Navbar
          fixed={true}
          button={false}
          getEvents={this.props.getEvents}
          totalPages={this.props.pageCount}
        />
        <br />
        <br />
        <br />
        <br />
        <Container style={{ textAlign: "center" }}>
          <Pagination
            firstItem={null}
            lastItem={null}
            activePage={activePage}
            pointing
            secondary
            siblingRange={2}
            onPageChange={this.handleChange}
            totalPages={this.props.pageCount || 20}
            id="pagination"
          />
          <div id="navigation">
            <Button
              content="Previous"
              icon="left arrow"
              labelPosition="left"
              onClick={this.handleBackClick}
              disabled={this.state.activePage > 1 ? null : true}
            />
            <Button
              content="Next"
              icon="right arrow"
              labelPosition="right"
              onClick={this.handleNextClick}
              disabled={this.state.activePage < 200 ? null : true}
            />
          </div>
          {this.props.loading ? (
            <Loading />
          ) : (
            <Container id="cards">
              <br />
              <br />
              {this.props.events.map((event, i) => (
                <EventCard event={event} key={i} />
              ))}
            </Container>
          )}
        </Container>
      </div>
    );
  }
}

export default App;
