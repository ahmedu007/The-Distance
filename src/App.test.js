import React from "react";
import ReactDOM from "react-dom";
import App from "./components/App";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<App events={[]} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("renders the loading state if there are no events", () => {
  const loading = renderer.create(<App loading={false} events={[]} />);
  expect(loading).toMatchSnapshot();
});
