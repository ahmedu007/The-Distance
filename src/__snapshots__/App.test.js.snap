// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the loading state if there are no events 1`] = `
<div>
  <div
    className="ui large top fixed menu"
  >
    <div
      className="ui container"
    >
      <a
        className="active item"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
      <div
        className="right item"
        onClick={[Function]}
      >
        <form
          onSubmit={[Function]}
        >
          <div
            className="ui left icon input"
          >
            <input
              disabled={undefined}
              onChange={[Function]}
              placeholder="Go to page..."
              tabIndex={undefined}
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="copy icon"
            />
          </div>
        </form>
      </div>
    </div>
  </div>
  <br />
  <br />
  <br />
  <br />
  <div
    className="ui container"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <div
      aria-label="Pagination Navigation"
      className="ui pagination pointing secondary menu"
      id="pagination"
      role="navigation"
    >
      <a
        aria-current={false}
        aria-label="Previous item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        value={NaN}
      >
        ⟨
      </a>
      <a
        aria-current={false}
        aria-label={undefined}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        value={1}
      >
        1
      </a>
      <a
        aria-current={false}
        aria-label={undefined}
        className="disabled item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        value={NaN}
      >
        ...
      </a>
      <a
        aria-current={false}
        aria-label={undefined}
        className="disabled item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        value={NaN}
      >
        ...
      </a>
      <a
        aria-current={false}
        aria-label={undefined}
        className="disabled item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={-1}
        value={NaN}
      >
        ...
      </a>
      <a
        aria-current={false}
        aria-label={undefined}
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        value={20}
      >
        20
      </a>
      <a
        aria-current={false}
        aria-label="Next item"
        className="item"
        onClick={[Function]}
        onKeyDown={[Function]}
        tabIndex={0}
        value={NaN}
      >
        ⟩
      </a>
    </div>
    <div
      id="navigation"
    >
      <button
        className="ui icon disabled left labeled button"
        disabled={true}
        onClick={[Function]}
        role="button"
        tabIndex={-1}
      >
        <i
          aria-hidden="true"
          className="left arrow icon"
        />
        Previous
      </button>
      <button
        className="ui icon disabled right labeled button"
        disabled={true}
        onClick={[Function]}
        role="button"
        tabIndex={-1}
      >
        <i
          aria-hidden="true"
          className="right arrow icon"
        />
        Next
      </button>
    </div>
    <div
      className="ui container"
      id="cards"
    >
      <br />
      <br />
    </div>
  </div>
</div>
`;
